core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/source/sql/asterixdb/AsterixDBSourceOpExec.scala

```diff
@@ -201,15 +205,49 @@ class AsterixDBSourceOpExec private[asterixdb] (
     * @throws IllegalArgumentException if attribute does not support string based search
     */
   @throws[IllegalArgumentException]
-  def addKeywordSearch(queryBuilder: StringBuilder): Unit = {
+  def addFilterConditions(queryBuilder: StringBuilder): Unit = {
+    if (search.getOrElse(false) && searchByColumn.isDefined && keywords.isDefined) {
+      val columnType = schema.getAttribute(searchByColumn.get).getType
 
-    val columnType = schema.getAttribute(searchByColumn.get).getType
+      if (columnType == AttributeType.STRING) {
 
-    if (columnType == AttributeType.STRING) {
+        queryBuilder ++= " AND ftcontains(" + searchByColumn.get + ", " + keywords.get + ") "
+      } else
+        throw new IllegalArgumentException("Can't do keyword search on type " + columnType.toString)
+    }
+
+    if (searchByColumnForRegex.isDefined && regex.isDefined) {
+      val regexColumnType = schema.getAttribute(searchByColumnForRegex.get).getType
+      if (regexColumnType == AttributeType.STRING) {
+        queryBuilder ++= "AND regexp_contains(" + searchByColumnForRegex.get + ", \"" + regex.get + "\") "
+      } else
+        throw new IllegalArgumentException(
+          "Can't do regex search on type " + regexColumnType.toString
+        )
+    }
 
-      queryBuilder ++= " AND ftcontains(" + searchByColumn.get + ", " + keywords.get + ") "
-    } else
-      throw new IllegalArgumentException("Can't do keyword search on type " + columnType.toString)
+    // geolocation must contain more than 1 points to from a rectangle or polygon
+    if (geolocation.size > 1 && geoAttributes.nonEmpty) {
+      val shape = {
+        val points = geolocation.flatMap(s => s.split(",").map(sub => sub.toDouble))
+        if (geolocation.size == 2) {
+          "create_rectangle(create_point(%.6f,%.6f), create_point(%.6f,%.6f))".format(points: _*)
+        } else {
+          "create_polygon([" + points.map(x => "%.6f".format(x)).mkString(",") + "])"
+        }
+      }
+      queryBuilder ++= " AND ("
+      var first = true
+      geoAttributes.foreach { attr =>
+        if (first) {
+          first = false
+        } else {
+          queryBuilder ++= " OR "
+        }
+        queryBuilder ++= s"spatial_intersect($attr, $shape)"
+      }
+      queryBuilder ++= " ) "
```

Comment:
I see you are joining the `spartial_intersect`, maybe just use `mkString`?

Reply:
`mkString` is hard to use in this case IMO.

Reply:
```suggestion
 geoAttributes.map{ attr =>  s"spatial_intersect($attr, $shape)"} .mkString(" OR ") 
```

this is clear right?

Reply:
true